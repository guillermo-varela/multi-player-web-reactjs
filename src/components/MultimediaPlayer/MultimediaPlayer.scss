.container-resource video, .container-resource audio, .container-resource img {
  width: 100%;
}
