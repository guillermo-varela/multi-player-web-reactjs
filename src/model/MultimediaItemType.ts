enum MultimediaItemType {
  AUDIO,
  IMAGE,
  VIDEO
}

export default MultimediaItemType;
