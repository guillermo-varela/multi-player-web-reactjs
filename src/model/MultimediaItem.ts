import MultimediaItemType from './MultimediaItemType';

export default class MultimediaItem {
  title: string;
  previewUrl: string;
  downloadUrl: string;
  type: MultimediaItemType;

  constructor(title: string, previewUrl: string, downloadUrl: string, type: MultimediaItemType) {
    this.title = title;
    this.previewUrl = previewUrl;
    this.downloadUrl = downloadUrl;
    this.type = type;
  }
}
