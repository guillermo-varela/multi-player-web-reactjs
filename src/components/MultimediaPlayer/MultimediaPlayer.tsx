import React from 'react';
import MultimediaItem from '../../model/MultimediaItem';
import MultimediaItemType from '../../model/MultimediaItemType';

import './MultimediaPlayer.scss';

type ItemToPlay = {
  item: MultimediaItem;
  index: number;
}

const MultimediaPlayer: React.FC<ItemToPlay> = ({ item, index }) => {
  if (item.type === MultimediaItemType.IMAGE) {
    return (
      <div>
        <h1>{item.title}</h1>
        <div className="container-resource">
          <img alt={item.title} src={item.downloadUrl} />
        </div>
      </div>
    );
  } else {
    return (
      <div>
        <h1>{item.title}</h1>
        <div className="container-resource">
          <video src={item.downloadUrl} poster={item.previewUrl} controls preload="none" />
        </div>
      </div>
    );
  }
}

export default MultimediaPlayer;
