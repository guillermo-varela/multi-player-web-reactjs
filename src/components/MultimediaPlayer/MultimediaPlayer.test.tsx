import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MultimediaPlayer from './MultimediaPlayer';
import MultimediaItem from '../../model/MultimediaItem';
import MultimediaItemType from '../../model/MultimediaItemType';

describe('<MultimediaPlayer />', () => {
  describe('for images', () => {
    const mockItem: MultimediaItem = {
      title: 'test',
      type: MultimediaItemType.IMAGE,
      previewUrl: 'www.preview.com',
      downloadUrl: 'www.download.com'
    }

    it('renders item title', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const titleContainerWrapper = wrapper.find('h1');
      expect(titleContainerWrapper).toHaveLength(1);
      expect(titleContainerWrapper.text()).toBe(mockItem.title);
    });

    it('renders item container', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const itemContainerWrapper = wrapper.find('div.container-resource');
      expect(itemContainerWrapper).toHaveLength(1);
    });

    it('renders <img />', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const imgWrapper = wrapper.find(`img[src="${mockItem.downloadUrl}"][alt="${mockItem.title}"]`);
      expect(imgWrapper).toHaveLength(1);
    });

    it('matches snapshot', () => {
      const tree = renderer.create(<MultimediaPlayer item={mockItem} index={1} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('for videos', () => {
    const mockItem: MultimediaItem = {
      title: 'test',
      type: MultimediaItemType.VIDEO,
      previewUrl: 'www.preview.com',
      downloadUrl: 'www.download.com'
    }

    it('renders item title', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const titleContainerWrapper = wrapper.find('h1');
      expect(titleContainerWrapper).toHaveLength(1);
      expect(titleContainerWrapper.text()).toBe(mockItem.title);
    });

    it('renders item container', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const itemContainerWrapper = wrapper.find('div.container-resource');
      expect(itemContainerWrapper).toHaveLength(1);
    });

    it('renders <video />', () => {
      const wrapper = shallow(<MultimediaPlayer item={mockItem} index={1} />);
      const imgWrapper = wrapper.find(`video[src="${mockItem.downloadUrl}"][poster="${mockItem.previewUrl}"][controls][preload="none"]`);
      expect(imgWrapper).toHaveLength(1);
    });

    it('matches snapshot', () => {
      const tree = renderer.create(<MultimediaPlayer item={mockItem} index={1} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
