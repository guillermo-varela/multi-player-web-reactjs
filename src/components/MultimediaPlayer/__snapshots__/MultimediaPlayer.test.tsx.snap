// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultimediaPlayer /> for images matches snapshot 1`] = `
<div>
  <h1>
    test
  </h1>
  <div
    className="container-resource"
  >
    <img
      alt="test"
      src="www.download.com"
    />
  </div>
</div>
`;

exports[`<MultimediaPlayer /> for videos matches snapshot 1`] = `
<div>
  <h1>
    test
  </h1>
  <div
    className="container-resource"
  >
    <video
      controls={true}
      poster="www.preview.com"
      preload="none"
      src="www.download.com"
    />
  </div>
</div>
`;
